gpt_index/vector_stores/pinecone.py

```diff
@@ -123,6 +168,22 @@ def __init__(
             self._query_kwargs = query_kwargs or {}
             self._delete_kwargs = delete_kwargs or {}
 
+        self._add_sparse_vector = add_sparse_vector
+        if tokenizer is None:
+            from transformers import BertTokenizerFast
+
+            orig_tokenizer = BertTokenizerFast.from_pretrained("bert-base-uncased")
+            # set some default arguments, so input is just a list of strings
+            tokenizer = partial(
+                orig_tokenizer,
+                padding=True,
+                truncation=True,
+                max_length=512,
+                # taking this out because causing errors
+                # special_tokens=False,
+            )
+        self._tokenizer = tokenizer
```

Comment:
It's not clear to me why tokenizer (and tokenization logic) should live inside `PineconeVectorStore`. IMO, we might want to keep the `VectorStore` abstraction simple, and keep the tokenizer logic one layer up (i.e. to be managed by the caller). This has the added benefit of sharing the tokenization logic across different vector store implementations.

Is it possible to pull this up into `GPTVectorStoreIndex` layer?

Reply:
Yeah tbh I was just following this: https://www.pinecone.io/learn/hybrid-search-intro/

It's a good question, right now i was thinking of computing sparse vectors within the vector store, b/c diff interfaces have different entry points. e.g. Weaviate just has you pass in the raw text (they might handle the sparse vectorization under the hood) 


What are your thoughts?

Reply:
Gotcha, I did not consider that different vector indices might handle sparse vectorization under the hood. In this case, I'm on board, and we can always generalize the tokenization logic when we need to add it for another vector store.

Could we at least pull the tokenizer construction logic out into a util method like e.g. `get default tokenizer` ?

Reply:
yep! that sounds good to me